CRM/MembershipExtras/Hook/PostProcess/UpdateSubscription.php

```diff
@@ -75,11 +75,30 @@ private function updateRecurringContribution() {
 
     if ($this->isUpdatedCycleDate()) {
       $params['next_sched_contribution_date'] = $this->nextContributionDate;
+      $params['start_date'] = $this->calculateNewStartDate();
     }
 
     civicrm_api3('ContributionRecur', 'create', $params);
   }
 
+  /**
+   * Calculates the new start date for the recurring contribution.
+   *
+   * Uses cycle day to calculate the start date for the contribution.
+   *
+   * @return string
+   */
+  private function calculateNewStartDate() {
+    $formValues = $this->form->exportValues();
```

Comment:
I left a note on the ticket saying that the frequency unit should also be taken into the account when doing the start date calculation from the cycle day.

Reply:
Dates are now calculated using the dates calculator, which uses cycle day.